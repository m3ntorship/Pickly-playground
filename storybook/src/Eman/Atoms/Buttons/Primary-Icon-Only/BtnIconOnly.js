import React from "react";
import PropTypes from "prop-types";

export const BUTTON_OPTIONS = {
  SIZE: {
    SMALL: "small",
    MEDIUM: "medium",
    BIG: "big",
  },
};

export const ButtonIconOnly = ({ size }) => {
  const IconSize =
    size === BUTTON_OPTIONS.SIZE.BIG
      ? "32"
      : size === BUTTON_OPTIONS.SIZE.MEDIUM
      ? "24"
      : "16";

  return (
    <button className="flex items-center rounded-full  bg-primary  text-white hover:bg-primary-shd8 focus:outline-none focus:border-2 focus:border-primary-shd9 focus:border-opacity-100 disabled:opacity-5">
      <svg
        width={IconSize}
        height={IconSize}
        viewBox="0 0 24 24"
        fill="none"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M16.293 9.293L12 13.586L7.70697 9.293L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.293Z"
          fill="white"
        />
      </svg>
    </button>
  );
};

ButtonIconOnly.propTypes = {
  size: PropTypes.string,
};
