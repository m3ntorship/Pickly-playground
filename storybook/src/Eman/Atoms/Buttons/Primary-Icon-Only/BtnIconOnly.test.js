import React from "react";
import renderer from "react-test-renderer";
import { ButtonIconOnly } from "./BtnIconOnly";

describe("Testing ButtonIconOnly", () => {
  it("Testing ButtonIconOnly size big", () => {
    const tree = renderer.create(<ButtonIconOnly size="big" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing ButtonIconOnly size medium", () => {
    const tree = renderer.create(<ButtonIconOnly size="medium" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing ButtonIconOnly size small", () => {
    const tree = renderer.create(<ButtonIconOnly size="small" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("Testing ButtonIconOnly with no props", () => {
    const tree = renderer.create(<ButtonIconOnly />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
