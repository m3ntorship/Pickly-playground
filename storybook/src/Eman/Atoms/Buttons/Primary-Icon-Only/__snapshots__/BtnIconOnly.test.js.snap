// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ButtonIconOnly Testing ButtonIconOnly size big 1`] = `
<button
  className="flex items-center rounded-full  bg-primary  text-white hover:bg-primary-shd8 focus:outline-none focus:border-2 focus:border-primary-shd9 focus:border-opacity-100 disabled:opacity-5"
>
  <svg
    fill="none"
    height="32"
    viewBox="0 0 24 24"
    width="32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16.293 9.293L12 13.586L7.70697 9.293L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.293Z"
      fill="white"
    />
  </svg>
</button>
`;

exports[`Testing ButtonIconOnly Testing ButtonIconOnly size medium 1`] = `
<button
  className="flex items-center rounded-full  bg-primary  text-white hover:bg-primary-shd8 focus:outline-none focus:border-2 focus:border-primary-shd9 focus:border-opacity-100 disabled:opacity-5"
>
  <svg
    fill="none"
    height="24"
    viewBox="0 0 24 24"
    width="24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16.293 9.293L12 13.586L7.70697 9.293L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.293Z"
      fill="white"
    />
  </svg>
</button>
`;

exports[`Testing ButtonIconOnly Testing ButtonIconOnly size small 1`] = `
<button
  className="flex items-center rounded-full  bg-primary  text-white hover:bg-primary-shd8 focus:outline-none focus:border-2 focus:border-primary-shd9 focus:border-opacity-100 disabled:opacity-5"
>
  <svg
    fill="none"
    height="16"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16.293 9.293L12 13.586L7.70697 9.293L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.293Z"
      fill="white"
    />
  </svg>
</button>
`;

exports[`Testing ButtonIconOnly Testing ButtonIconOnly with no props 1`] = `
<button
  className="flex items-center rounded-full  bg-primary  text-white hover:bg-primary-shd8 focus:outline-none focus:border-2 focus:border-primary-shd9 focus:border-opacity-100 disabled:opacity-5"
>
  <svg
    fill="none"
    height="16"
    viewBox="0 0 24 24"
    width="16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M16.293 9.293L12 13.586L7.70697 9.293L6.29297 10.707L12 16.414L17.707 10.707L16.293 9.293Z"
      fill="white"
    />
  </svg>
</button>
`;
